driver-core/src/main/com/mongodb/operation/CreateIndexesOperation.java

```diff
@@ -177,7 +177,7 @@ private BsonDocument getIndex(final IndexRequest request) {
         if (request.isUnique()) {
             index.append("unique", BsonBoolean.TRUE);
         }
-        if (request.isSparse()) {
+        if (request.isSparse() && request.getPartialFilterExpression() == null) {
```

Comment:
I've updated SPEC-334

> You cannot specify both the partialFilterExpression option and the sparse option

As a partial filter is a superset of the functionality of a sparse index - I opted to just not send the sparse flag in this situation.  Not sure why that is problematic as such but I can leave it to fall through and then service the server error, but thats probably  not the best UX though.